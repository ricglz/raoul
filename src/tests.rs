use super::{parse, DirFunc};

#[test]
fn use_of_undeclared_variable() {
    let filename = "examples/invalid/undeclared-variable.ra";
    let program = std::fs::read_to_string(filename).expect(filename);
    let debug = true;
    let ast_response = parse(&program, debug);
    assert!(ast_response.is_ok());
    let ast = ast_response.unwrap();
    let mut dir_func = DirFunc::new();
    let build_res = dir_func.build_dir_func(ast);
    assert!(build_res.is_err());
}
