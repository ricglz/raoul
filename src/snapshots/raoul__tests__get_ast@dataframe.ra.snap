---
source: src/tests.rs
expression: ast
input_file: src/examples/valid/dataframe.ra
---
Main(([], [], [
    Assignment(false, Id(dataframe), ReadCSV(String(song_data_clean.csv))),
    Assignment(false, Id(rows), PureDataframeOp(Rows, dataframe)),
    Assignment(false, Id(columns), PureDataframeOp(Columns, dataframe)),
    Write([Id(rows), Id(columns)]),
    Assignment(false, Id(col), String(danceability)),
    Write([UnaryDataframeOp(Average, dataframe, String(song_duration_ms))]),
    Write([UnaryDataframeOp(Std, dataframe, Id(col))]),
    Write([UnaryDataframeOp(Median, dataframe, Id(col))]),
    Write([UnaryDataframeOp(Variance, dataframe, Id(col))]),
    Write([UnaryDataframeOp(Min, dataframe, Id(col))]),
    Write([UnaryDataframeOp(Max, dataframe, Id(col))]),
    Write([UnaryDataframeOp(Range, dataframe, Id(col))]),
    Write([Correlation(dataframe, Id(col), String(song_duration_ms))]),
]))
